library/src/main/java/com/novoda/downloadmanager/notifications/NotificationDisplayer.java

```diff
@@ -116,39 +107,60 @@ private void buildIcon(int type, NotificationCompat.Builder builder) {
         }
     }
 
-    private void buildActionIntents(String tag, int type, Collection<DownloadBatch> cluster, NotificationCompat.Builder builder) {
+    private void buildActionIntents(int type, Collection<DownloadBatch> cluster, NotificationCompat.Builder builder) {
         DownloadBatch batch = cluster.iterator().next();
         long batchId = batch.getBatchId();
         int batchStatus = batch.getStatus();
 
         if (type == SynchronisedDownloadNotifier.TYPE_ACTIVE || type == SynchronisedDownloadNotifier.TYPE_WAITING) {
-            // build a synthetic uri for intent identification purposes
-            Uri uri = new Uri.Builder().scheme("active-dl").appendPath(tag).build();
-
-            Intent clickIntent = createClickIntent(ACTION_LIST, batchId, batchStatus, uri); // TODO: the Uri doesn't seem to be ever read
-            builder.setContentIntent(PendingIntent.getBroadcast(context, 0, clickIntent, PendingIntent.FLAG_UPDATE_CURRENT));
             builder.setOngoing(true);
-
         } else if (type == SynchronisedDownloadNotifier.TYPE_SUCCESS
                 || type == SynchronisedDownloadNotifier.TYPE_CANCELLED
                 || type == SynchronisedDownloadNotifier.TYPE_FAILED) {
-            long firstDownloadBatchId = batch.getFirstDownloadBatchId(); // TODO why can't we just use getBatchId()?
-            Uri uri = ContentUris.withAppendedId(downloadsUriProvider.getAllDownloadsUri(), firstDownloadBatchId);
-
-            Intent hideIntent = new Intent(ACTION_HIDE, uri, context, DownloadReceiver.class);
-            hideIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batchId);
-            builder.setDeleteIntent(PendingIntent.getBroadcast(context, 0, hideIntent, 0));
-
+            Intent dismissedIntent = createNotificationDismissedIntent(batchId);
+            builder.setDeleteIntent(PendingIntent.getBroadcast(context, 0, dismissedIntent, 0));
             builder.setAutoCancel(true);
-
-            String action = batch.isError() ? ACTION_LIST : ACTION_OPEN;
-            Intent clickIntent = createClickIntent(action, batchId, batchStatus, uri);  // TODO: the Uri doesn't seem to be ever read
-            builder.setContentIntent(PendingIntent.getBroadcast(context, 0, clickIntent, PendingIntent.FLAG_UPDATE_CURRENT));
         }
 
+        Intent clickIntent = createClickIntent(batchId, batchStatus);
```

Comment:
the click intent is needed in all cases I believe, so instead of doing it in the if AND the else, it's done here now